src/astra_driver.cpp

```diff
@@ -408,37 +411,42 @@ void AstraDriver::colorConnectCb()
   //color_subscribers_ = pub_color_.getNumSubscribers() > 0;
 
   //if (color_subscribers_ && !device_->isColorStreamStarted())
-  if (!device_->isIRStreamStarted())
+  if (!device_->isColorStreamStarted())
   {
-    // Can't stream IR and RGB at the same time. Give RGB preference.
-    if (device_->isIRStreamStarted())
+    if(can_publish_color_)
     {
-      ROS_ERROR("Cannot stream RGB and IR at the same time. Streaming RGB only.");
-      ROS_INFO("Stopping IR stream.");
-      device_->stopIRStream();
-    }
-
-    device_->setColorFrameCallback(boost::bind(&AstraDriver::newColorFrameCallback, this, _1));
+      // Can't stream IR and RGB at the same time. Give RGB preference.
+      if (device_->isIRStreamStarted())
+      {
+        ROS_ERROR("Cannot stream RGB and IR at the same time. Streaming RGB only.");
+        ROS_INFO("Stopping IR stream.");
+        device_->stopIRStream();
+      }
 
-    ROS_INFO("Starting color stream.");
-    device_->startColorStream();
+      device_->setColorFrameCallback(boost::bind(&AstraDriver::newColorFrameCallback, this, _1));
 
+      ROS_INFO("Starting color stream.");
+      device_->startColorStream();
+    }
+    else
+    {
+      ROS_INFO("Attempted to start RGB stream, but RGB streaming was disabled.");
+    }
   }
   //else if (!color_subscribers_ && device_->isColorStreamStarted())
   else if (device_->isColorStreamStarted())
   {
     ROS_INFO("Stopping color stream.");
     device_->stopColorStream();
 
+    // TODO: With the IR subscriber check commented out, this section just
```

Comment:
:+1: we don't run linters here but I think it's good to comply with the conventions to minimize the work in the future to make it comply